test/lint/import-resolver.js

```diff
@@ -0,0 +1,31 @@
+/**
+ * External dependencies
+ */
+const resolverNode = require( 'eslint-import-resolver-node' );
+
+exports.interfaceVersion = 2;
+
+exports.resolve = function ( source, file, config ) {
+	if ( source.startsWith( '@wordpress/' ) ) {
+		const packageName = source.slice( '@wordpress/'.length );
+
+		let entryFile;
+		try {
+			entryFile = require.resolve( `../../packages/${ packageName }` );
+		} catch {
+			entryFile = require.resolve(
+				`../../packages/${ packageName }/src/index.js`
```

Comment:
Can it omit the file name? I theory, the entry point could be now `index.ts` or even less likely but possible `index.tsx`.